python/ray/data/dataset.py

```diff
@@ -101,20 +101,27 @@
 
 @PublicAPI
 class Dataset(Generic[T]):
-    """Implements a distributed Arrow dataset.
+    """A Dataset is a distributed data collection for data loading and processing.
 
-    Datasets are implemented as a list of ``ObjectRef[Block]``. The block
-    also determines the unit of parallelism. The default block type is the
-    ``pyarrow.Table``. Arrow-incompatible objects are held in ``list`` blocks.
+    Datasets are implemented as a list of ``ObjectRef[Block]``, where each block
+    holds an ordered collection of items, representing a segment of the overall
```

Comment:
```suggestion
    holds an ordered collection of items, representing a partition of the overall
```

Reply:
"block" is basically the same as "partition" for us, and is more accurate here than "segment"

Reply:
It's not that clear in our context -- we have things named as "BlockPartition", which still take me a moment to read it.

Changed it to "shard", which I believe is accurate and canonical in distributed data systems.

Reply:
"shard" sounds good!

I wish that we had called "blocks" --> "partitions" from the start, especially given our leaning towards tabular data.